qpid-docker/README.md

```diff
@@ -0,0 +1,198 @@
+This folder belongs to user qpid, which is part of the root group. Java process is executed under the qpid user as well.
+
+### Running the Container
+
+Container can be started using following command:
+```
+docker run -d -p 5672:5672 -p 8080:8080 --name qpid <IMAGE_NAME>
+```
+There are two ports exposed: 5672 for AMQP connections and 8080 for HTTP connections.
+
+There are following environment variables available when running the container:
+
+| Environment Variable | Description                                                                  |
+|----------------------|------------------------------------------------------------------------------|
+| JAVA_GC              | JVM Garbage Collector parameters, default value "-XX:+UseG1GC"               |
+| JAVA_MEM             | JVM memory parameters, default value "-Xmx300m -XX:MaxDirectMemorySize=200m" |
+| JAVA_OPTS            | Further JVM parameters, empty by default                                     |
+
+#### Container Volume
+
+Before starting the container a volume should be created:
+
+```
+docker volume create --driver local --opt device=<PATH_TO_FOLDER_ON_HOST_MACHINE> --opt type=none --opt o=bind qpid_volume
+```
+
+Then container using created volume can be started as follows:
+
+```
+docker run -d -p 5672:5672 -p 8080:8080 -v qpid_volume:/qpid-broker-j/work --name qpid <IMAGE_NAME>
+```
+or
+```
+podman run -d -p 5672:5672 -p 8080:8080 -v qpid_volume:/qpid-broker-j/work:Z --name qpid <IMAGE_NAME>
+```
+When container is started for the first time, configuration files become available in the mapped folder being persisted 
+after subsequent restarts.
+
+### Broker Users
+
+Default configuration provides two preconfigured broker users:
+- admin (password 'admin')
+- guest (password 'guest')
+
+User 'admin' has read and write access to all broker objects.
+User 'guest' has read-only access to all broker objects, can not send or consume messages.
+
+Further broker users as well as other broker objects (queues, exchanges, keystores, truststore, ports etc.)
+can be created via HTTP management interface. Description of the broker REST API can be found in broker book (chapter 6.3).
+
+To change user password following command can be used:
+
+```
+curl -d '{"password": "<NEW_PASSWORD>"}' http://admin:admin@localhost:8080/api/latest/user/plain/<USERNAME>
+```
+
+## Broker Customization
+
+To customize broker before building the container image, its configuration files may be edited to start broker with
+queues, exchanges, users or other objects.
+
+The file config.json contains definitions of the broker objects and references a file containing definitions 
+of virtualhost objects (exchanges and queues).
+
+It may be helpful first to create broker objects needed via broker web GUI or via REST API, and then investigate the 
+configuration files and copy the appropriate definitions to the configuration files used for container image creation.
```

Comment:
I'd expect detail around this to be in/referenced-by the user manual bits as commented previously.

This still seems tailored to only customising the default image config before creation though and building it with this module. Users should just have a way they can supply replacement config files to be used by the existing image on first start, not need to track down the source release archive for the image they start with in order to get this module and modify it and build a new image.

E.g the activemq-artemis image lets people provide override files by providing them in a directory they will be copied from, over top of the default files that were just created: https://github.com/apache/activemq-artemis/blob/2.31.2/artemis-docker/docker-run.sh#L38-L40. That way users can use the existing image but configure things how they like. That feature was contributed by a user in https://github.com/apache/activemq-artemis/pull/4575 after the first Artemis image was published, which behaved a lot like this one currently would seem to.